crates/wash-lib/src/start/github.rs

```diff
@@ -87,3 +90,37 @@ where
         .await
         .map_or(false, |m| m.is_file())
 }
+
+/// Helper function to set up a reqwest client for performing the download
+pub(crate) fn get_download_client() -> Result<reqwest::Client> {
+    let proxy_username = std::env::var("WASH_PROXY_USERNAME").unwrap_or_default();
+    let proxy_password = std::env::var("WASH_PROXY_PASSWORD").unwrap_or_default();
+    let http_proxy_addr = std::env::var("HTTP_PROXY")
+        .or_else(|_| std::env::var("http_proxy"))
+        .unwrap_or_default();
+    let https_proxy_addr = std::env::var("HTTPS_PROXY")
+        .or_else(|_| std::env::var("https_proxy"))
+        .unwrap_or_default();
```

Comment:
nit: I think keeping these as `Option<String>`'s will keep the code a little cleaner, and then later using an `if let` would work well.
```rust
if let Some(http_proxy) = http_proxy_addr {}
```

Reply:
You could even fetch the env var in the same statement if you don't need it after this block
```rust
if let Some(http_proxy_addr) = std::env::var("HTTP_PROXY")
        .or_else(|_| std::env::var("http_proxy")) {}
```